smarts/core/vehicle_index.py

```diff
@@ -230,29 +230,34 @@ def switch_control_to_agent(
         self, sim, vehicle_id, agent_id, boid=False, recreate=False
     ):
         self._log.debug(f"Switching control of {agent_id} to {vehicle_id}")
+        vehicle = None
         if recreate:
             # XXX: Recreate is presently broken for bubbles because it impacts the
             #      sumo traffic sim sync(...) logic in how it detects a vehicle as
             #      being hijacked vs joining. Presently it's still used for trapping.
             return self._switch_control_to_agent_recreate(
                 sim, vehicle_id, agent_id, boid
             )
-
-        vehicle = self._vehicles[vehicle_id]
-        ackermann_chassis = AckermannChassis(pose=vehicle.pose, bullet_client=sim.bc)
-        vehicle.swap_chassis(ackermann_chassis)
-
-        v_index = self._controlled_by["vehicle_id"] == vehicle_id
-        entity = _ControlEntity(*self._controlled_by[v_index][0])
-        self._controlled_by[v_index] = tuple(
-            entity._replace(
-                actor_type=_ActorType.Agent,
-                actor_id=agent_id,
-                shadow_actor_id="",
-                is_boid=boid,
+        else:
```

Comment:
I wouldn't nest this in the `else`, you've introduced a `vehicle=None` above because you reserve space in the traffic sim below (line 258), but vehicle is only set to a non-None value in this `else` branch.

Either de-indent back to how it was or move the traffic sim reservation + `return vehicle` into the else branch. Either way we need to remove the `vehicle=None` above

Reply:
The intention here was to forward the vehicle from `_switch_control_to_agent_recreate()` and accidentally got lost in a merge resolution.

I realise now this still did not accomplish what I was intending. A entry location should not be reserved when just taking over control of an existing vehicle.